File: 剑指offer/剑指 Offer 27. 二叉树的镜像.cpp
class Solution {
public:
    TreeNode* mirrorTree(TreeNode* root) {
        if(!root) return root;
        queue<TreeNode*> q;
        q.push(root);
    while(!q.empty()){
        TreeNode* node=q.front();
        q.pop();
        swap(node->left,node->right);
        if(node->left) q.push(node->left);
        if(node->right) q.push(node->right);
    }
    return root;
    }
};



class Solution {
public:
    TreeNode* mirrorTree(TreeNode* root) {
        if (root == NULL)   return NULL; 
        swap(root->left, root->right);
        mirrorTree(root->left);
        mirrorTree(root->right);
        
        return root;        
    }
};

File: 剑指offer_java/剑指 Offer 27. 二叉树的镜像.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode mirrorTree(TreeNode root) {
        if(root==null) return null;
        TreeNode x= root.right;
        root.right=root.left;
        root.left=x;
        mirrorTree(root.left);
        mirrorTree(root.right);
        return root;
    }
}
